web-common/src/features/dashboards/dimension-table/dimension-table-utils.ts

```diff
@@ -271,3 +215,229 @@ export function estimateColumnSizes(
+        sorted = sortDirection;
+      } else if (
+        name.endsWith("_delta_perc") &&
+        sortType === SortType.DELTA_PERCENT
+      ) {
+        sorted = sortDirection;
+      } else if (
+        name.endsWith("_percent_of_total") &&
+        sortType === SortType.PERCENT
+      ) {
+        sorted = sortDirection;
+      } else if (name === selectedMeasure.name && sortType === SortType.VALUE) {
+        sorted = sortDirection;
+      }
+
+      if (measureNames.includes(name)) {
+        // Handle all regular measures
+        const measure = allMeasures.find((m) => m.name === name);
+        return {
+          name,
+          type: "INT",
+          label: measure?.label || measure?.expression,
+          description: measure?.description,
+          total: referenceValues[measure.name] || 0,
+          enableResize: false,
+          format: measure?.format,
+          highlight,
+          sorted,
+        };
+      } else if (name === dimensionColumn) {
+        // Handle dimension column
+        return {
+          name,
+          type: "VARCHAR",
+          label: dimension?.label,
+          enableResize: true,
+          highlight,
+          sorted,
+        };
+      } else if (selectedMeasure) {
+        // Handle delta and delta_perc
+        const comparison = getComparisonProperties(name, selectedMeasure);
+        return {
+          name,
+          type: comparison.type,
+          label: comparison.component,
+          description: comparison.description,
+          enableResize: false,
+          format: comparison.format,
+          highlight,
+          sorted,
+        };
+      }
+      return undefined;
+    })
+    .filter((column) => !!column);
+}
+
+/**
+ * Splices the context column names into the list of dimension
+ * table column names.
+ *
+ * This mutates the columnNames array.
+ */
+export function addContextColumnNames(
+  columnNames: string[],
+  timeComparison: boolean,
+  validPercentOfTotal: boolean,
+  selectedMeasure: MetricsViewMeasure
+) {
+  const name = selectedMeasure?.name;
+
+  const sortByColumnIndex = columnNames.indexOf(name);
+  // Add comparison columns if available
+  let percentOfTotalSpliceIndex = 1;
+  if (timeComparison) {
+    percentOfTotalSpliceIndex = 2;
+    columnNames.splice(sortByColumnIndex + 1, 0, `${name}_delta`);
+
+    // Only push percentage delta column if selected measure is not a percentage
+    if (selectedMeasure?.format != FormatPreset.PERCENTAGE) {
+      percentOfTotalSpliceIndex = 3;
+      columnNames.splice(sortByColumnIndex + 2, 0, `${name}_delta_perc`);
+    }
+  }
+  if (validPercentOfTotal) {
+    columnNames.splice(
+      sortByColumnIndex + percentOfTotalSpliceIndex,
+      0,
+      `${name}_percent_of_total`
+    );
+  }
+}
+
+/**
+ * This function prepares the data for the dimension table
+ * from data returned by the createQueryServiceMetricsViewComparisonToplist
+ * API.
+ *
+ */
+export function prepareDimensionTableRows(
```

Comment:
Same as above

Reply:
Hey @AdityaHegde, regarding this comment and the other two: I'm doing more work on this stuff to enable sorting by dimension values (have a PR underway right now). Would it be ok to accept this PR for now and I can clean this up as you suggest in the follow-up PR, since I'm making changes to this code anyway?